pvlib/spectrum/mismatch.py

```diff
@@ -0,0 +1,213 @@
+    .. [1] ASTM "G173-03 Standard Tables for Reference Solar Spectral
+        Irradiances: Direct Normal and Hemispherical on 37° Tilted Surface."
+    '''
+    # Contributed by Anton Driesse (@adriesse), PV Performance Labs. Aug. 2022
+
+    pvlib_path = pvlib.__path__[0]
+    filepath = os.path.join(pvlib_path, 'data', 'astm_g173_am15g.csv')
+
+    am15g = pd.read_csv(filepath, index_col=0, squeeze=True)
+
+    if wavelength is not None:
+        interpolator = interp1d(am15g.index, am15g,
+                                kind='linear',
+                                bounds_error=False,
+                                fill_value=0.0,
+                                copy=False,
+                                assume_sorted=True)
+
+        am15g = pd.Series(data=interpolator(wavelength), index=wavelength)
+
+    am15g.index.name = 'wavelength'
+    am15g.name = 'am15g'
+
+    return am15g
+
+
+def calc_spectral_mismatch(sr, e_sun, e_ref=None):
+    """
+    Calculate spectral mismatch between a test device and broadband reference
+    device under specified solar spectral irradiance conditions.
+
+    Parameters
+    ----------
+    sr: pandas.Series
+        The relative spectral response of one (photovoltaic) test device.
+        The index of the Series must contain wavelength values in nm. [-]
+    e_sun: pandas.DataFrame or pandas.Series
+        One or more measured solar irradiance spectra in a pandas.DataFrame
+        having wavelength in nm as column index.  A single spectrum may be
+        be given as a pandas.Series having wavelength in nm as index.
+        [W/m^2/nm]
+    e_ref: pandas.Series, optional
+        The reference spectrum to use for the mismatch calculation.
+        The index of the Series must contain wavelength values in nm.
+        The default is the ASTM G173-03 global tilted spectrum. [W/m^2/nm]
+
+    Returns
+    -------
+    smm: pandas.Series or float if a single measured spectrum is provided. [-]
+
+    Notes
+    -----
+    If the default reference spectrum is used it is linearly interpolated
+    to the wavelengths of the measured spectrum.  To achieve alternate
+    behavior e_ref can be transformed before calling this function and
+    provided as an argument.
+
+    The spectral response is linearly interpolated to the wavelengths of the
+    spectrum with which is it multiplied internally (e_sun and e_ref). To
+    achieve alternate behavior the spectral response can be transformed
+    before calling this function.
+
+    The standards describing mismatch calculations focus on indoor laboratory
+    applications, but are applicable to outdoor performance as well.
+    The most recent version of ASTM E973 [1]_ is somewhat more difficult to
+    read because it includes adjustments for temperature dependency of the
+    spectral response, which led to a formulation using quantum efficiency
+    (QE). IEC 60904-7 is clearer and also discusses the use of a
+    broadband reference device.
```

Comment:
The temperature dependence of the SR would typically be relevant for PV devices deployed outside. NREL found this effect to be significant in their primary reference cell calibrations that are performed outdoors against broadband cavity radiometers. For other applications, perhaps the effect could be considered negligible.

Reply:
I don't think it's necessarily negligible, but the information needed for the calculation is even more unavailable than SR data at 25C.